### DISCUSSION TITLE
Custom Bundle step by step by docs

### DISCUSSION THREAD

**Discussion by nedelyux** [symfony/symfony#49975] [Q&A] [answered]
### Symfony version(s) affected

6.2

### Description

https://symfony.com/doc/current/bundles.html

On this documentation page they are trying to show us how to create our own bundle
I have completed everything step by step

1. Created a file in my project src/AcmeTestBundle.php

2. Put Acme\TestBundle\AcmeTestBundle::class => ['all' => true],
to config/bundles.php

3. Add "Acme\\TestBundle\\": "src/"
to composer in project folder

------

And couldn't launch the project

```
Symfony\Component\ErrorHandler\Error\ClassNotFoundError^ {#67
  #message: """
    Attempted to load class "AcmeTestBundle" from namespace "Acme\TestBundle".\n
    Did you forget a "use" statement for another namespace?
    """
  #code: 0
  #file: "./vendor/symfony/framework-bundle/Kernel/MicroKernelTrait.php"
  #line: 126

```

### How to reproduce

Follow the documentation

**Comment by ahundiak** [comment]
Did you do a `composer dump-autoload` after editing composer.json? 

If you are developing inside of a regular Symfony project then I find trying to put a bundle under src is a bit confusing.  I make another source directory for it.  In any event, here is a [simple working bundle](https://github.com/cerad/s62bundle).  Might help for comparison purposes.

**Reply by nedelyux** [reply]
No, I didn't...
I'm new to Symfony, and in general to PHP
But the fact is that the documentation does not show the creation of the bundle completely

Thanks for the example

**Reply by ahundiak** [reply]
Symfony docs are great except when they are not.  Bundles are considered to be an advanced topic so `skipping` a step or two involving PHP type details along the way is understandable.  And Symfony does accept pull request.  Especially for documentation.

**Comment by xabbuh** [comment]
Can you create a GitHub repository with your attempt please?

**Reply by nedelyux** [reply]
There's no need for that

You just need to enter symfony new [project_name] and you will get an empty application

I still didn't understand where to place the bundle (meaning in the project directory or outside it)

How to connect (meaning is it enough to edit config/bundles.php )

I didn't find these answers even on google

**Reply by ahundiak** [reply]
Did you happen to look at the link I gave you?  src-my-bundle might offer a clue.

There are actually several approaches to developing a bundle.  But it's basically a PHP tooling question.  Getting one bundle to work is a good first step.

**Comment by nedelyux** [comment, marked as answer]
### https://habr.com/ru/articles/498134/

I found a solution how to connect the bundle

It seems to me that my problem was in some of the ways

- I didn't make composer require [bundle-name]
- I didn't have "type": "symfony-bundle". And also the "description" or the "require": "php" or "extra"
fields - Or the "repositories" type: "path" fields in the composer.json root

In any case, none of these points are mentioned in the documentation....
